File: typescript/Behavioral/20-Strategy/Strategy/FixedRateStrategy.ts
import { IPricingStrategy } from "./IPricingStrategy";
import type { JobCard } from "../Model/JobCard";

export class FixedRateStrategy implements IPricingStrategy {
  readonly name = "Fixed Rate";

  calculate(job: JobCard): number {
    return job.quotedPrice;
  }

  explain(job: JobCard): string {
    return `Flat quoted price of N$${job.quotedPrice.toFixed(2)}, regardless of time taken.`;
  }
}

File: csharp/Behavioral/20-Strategy/Strategy/FixedRateStrategy.cs
namespace Strategy;

public class FixedRateStrategy : IPricingStrategy
{
    public string Name => "Fixed Rate";

    public decimal Calculate(JobCard job) => job.QuotedPrice;

    public string Explain(JobCard job) =>
        $"Flat quoted price of N${job.QuotedPrice:N2}, regardless of time taken.";
}
